src/analysis/object.rs

```diff
@@ -97,21 +82,16 @@ pub fn class(env: &Env, obj: &GObject, deps: &[library::TypeId]) -> Option<Info>
 
     let supertypes = supertypes::analyze(env, class_tid, &mut imports);
 
-    let mut generate_trait = obj.generate_trait;
+    let final_type = klass.final_type;
+    // We would like to generate a trait if this is not a final type as there can be
+    // subtypes, however this will be overridden again below if there is nothing that
```

Comment:
`will be overridden again below` is what I meant btw

Reply:
Sorry missed that, maybe remove this and use `!final_type` 3 middle call instead ?

Reply:
Ok, fine with me :) But then the comment makes no sense and there's no explanation